Update peerDependencies to support Serverless Framework v4

This plugin doesn't work with the Serverless Framework v4 due to the peerDependencies settings. Getting the following error on install:

```
plugins-tests » npm i 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: plugins-tests@1.0.0
npm ERR! Found: serverless@4.5.0
npm ERR! node_modules/serverless
npm ERR!   dev serverless@"^4.5.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer serverless@"3" from serverless-apigateway-service-proxy@2.2.1
npm ERR! node_modules/serverless-apigateway-service-proxy
npm ERR!   dev serverless-apigateway-service-proxy@"^2.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/eslam/.npm/_logs/2025-01-27T16_04_23_698Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/eslam/.npm/_logs/2025-01-27T16_04_23_698Z-debug-0.log
```

As a workaround until peerDependencies are updated, users can add the following to their `package.json` file:

```
"overrides": {
    "serverless-apigateway-service-proxy": {
      "serverless": "$serverless"
    }
  }
```

The workaround isn't working for me using using `serverless@4.5.4` as mentioned above or the latest (`serverless@4.9.1`)

```
✖ Error: Cannot find module 'serverless/lib/plugins/aws/package/compile/events/api-gateway/lib/rest-api'
Require stack:
- /Users/username/r/myrepo/node_modules/serverless-apigateway-service-proxy/lib/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
    at Module._load (node:internal/modules/cjs/loader:981:27)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/Users/username/r/myrepo/node_modules/serverless-apigateway-service-proxy/lib/index.js:7:24)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:203:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async PluginManager.requireServicePlugin (file:///Users/username/.serverless/releases/4.9.1/package/dist/sf-core.js:809:1948)
    at async PluginManager.resolveServicePlugins (file:///Users/username/.serverless/releases/4.9.1/package/dist/sf-core.js:809:2750)
    at async PluginManager.loadAllPlugins (file:///Users/username/.serverless/releases/4.9.1/package/dist/sf-core.js:809:1292)
```

--- a/package.json
+++ b/package.json
@@ -28,7 +28,7 @@
     "common-tags": "^1.8.0"
   },
   "peerDependencies": {
-    "serverless": "3"
+    "serverless": ">=3"
   },
   "jest": {
     "setupFilesAfterEnv": [

